torch/nn/attention/_flex_attention.py

```diff
@@ -472,12 +496,30 @@ def score_mod(
 
     """
 
+    if not query.size(-1) == key.size(-1):
+        raise ValueError(
+            "NYI: Embedding dimension of the query and key must be the same"
```

Comment:
This doesn't need to be a NYI - this is just always a requirement.

Reply:
Resolved.